### DISCUSSION TITLE
Using hair shader in another Plugin.

### DISCUSSION THREAD

**Discussion by JoseConseco** [soupday/cc3_blender_tools#184] [Q&A]
Hi @soupday ,  some of customers of my Hair Tool plugin, wish it comes with same hair shader as in yours CC_belnder addon. Is it ok if I used you shader in [Hair Tool](https://bartoszstyperek.gumroad.com/l/hairtool) ?  I'm not sure if you work with/for Reallusion, and If I should ask for permission from you or from them?  I collaborated with them already in past, where they included HairTool addon for each purchase of CC product. So I would hope, they would be ok with this request.

**Comment by soupday** [comment]
I don't have a problem with it. 

The shader is my own version of the Smart Hair shader in Character Creator/iClone, but as I wasn't given any particular specification to make it from so it doesn't work _exactly_ the same as the Reallusion one, but it's close enough visually.

As it was based on their shader and texture configuration I suppose they should have the final say about it.

Some things to note:
There are two versions of the shader, one for Eevee and one for Cycles. As Eevee doesn't do anisotropic specular highlights so the Eevee version fakes the highlights using the tangents based on the flow maps and the surface normals passed through a fresnel node.

It is also made to be controlled from the add-on UI rather than the material properties. As it stands it would be awkward to control without the add-ons UI.

If you want I could put together a version that combines the Eevee and Cycles and works better with the Material Properties tab.

**Comment by JoseConseco** [comment]
@soupday if you could do this, it would be great, but no pressure - I can rearrange inputs myself. I think, I'm using the same trick to get anisotropic spec working in my hair shader too.   Thanks again.

**Comment by soupday** [comment]
Here's a tidied up version, Eevee and Cycles in the same shader and all the material controls are available in the material properties panel.

[Hair.zip](https://github.com/soupday/cc_blender_tools/files/12498122/Hair.zip)

**Comment by JoseConseco** [comment]
Wow, that is way cleaner. The base  setup as quite confusing.   I'm not sure if you are doing any 3d, but if you do, then just send me message and I can give you any of [my addons](https://bartoszstyperek.gumroad.com) for free. 
A bet all these controls will look scary to the users, but that is what they wanted ;)
